pkg/network/tracer/testutil/gotls_client/gotls_client.go

```diff
@@ -35,8 +35,14 @@ func main() {
 		},
 	}
 
+	in := make([]byte, 1)
```

Comment:
Do we need this?

Reply:
Well I only use it for ReadFull to have a buffer to read to, since I use it to wait for something to happen on stdin. Is there a better way to wait for data on Stdin?

Reply:
Sorry I mean do we need this whole new code block (including ReadFull)? Is this just for a delay? The code below sends a byte right after start, so I'm not sure what this is buying us.

Reply:
I see. So the error would not be here, but rather in the test for hooking already running processes. In this test, we want to start the client before the tracer, but have it wait a bit before sending the requests. It worked because of the 5 seconds sleep just below (we don't have a way to be notified when process hooking finishes without using reflection so it will have to stay here or be moved in the test body (or we use reflection, which I think is acceptable in a testing context)).

For now, I just pushed a fix to make the client send requests after the tracer is started in the "already running processes" test.